luarules/gadgets/game_tax_resource_sharing.lua

```diff
@@ -56,27 +75,39 @@ function gadget:AllowResourceTransfer(senderTeamId, receiverTeamId, resourceType
 	-- rShare is the share slider setting, don't exceed their share slider max when sharing
 	local maxShare = rStor * rShare - rCur
 
-	local taxedAmount = math.min((1-sharingTax)*amount, maxShare)
-	local totalAmount = taxedAmount / (1-sharingTax)
-	local transferTax = totalAmount * sharingTax
+	-- Prevent negative maxShare
+	maxShare = math.max(0, maxShare)
 
-	Spring.SetTeamResource(receiverTeamId, resourceName, rCur+taxedAmount)
-	local sCur, _, _, _, _, _ = Spring.GetTeamResources(senderTeamId, resourceName)
-	Spring.SetTeamResource(senderTeamId, resourceName, sCur-totalAmount)
+	local transferAmount = math.min(amount, maxShare)
 
-	-- Block the original transfer
-	return false
-end
+	local currentCumulative = cumulativeMetalSent[senderTeamId] or 0
+	local breakdown = Tax.computeTransfer(resourceName, transferAmount, sharingTax, metalTaxThreshold, currentCumulative)
+	local actualSentAmount = breakdown.actualSent
+	local actualReceivedAmount = breakdown.actualReceived
```

Comment:
I think I messed up my cherry-pick here and this needs some love because it's missing an end. I've cherry-picked this too many times and swear I tested this but it's been a minute so let me know if anyone looks at this again. The logic is solid here I'm using it downstream so just let me know and I'll fix this branch up